app/models/csv_export.rb

```diff
@@ -0,0 +1,79 @@
+class CsvExport < ActiveRecord::Base
+  include Searchable
+
+  belongs_to :user
+
+  def ready?
+    finished? or downloaded?
+  end
+
+  def pending?
+    status == 'pending'
+  end
+
+  def started?
+    status == 'started'
+  end
+
+  def finished?
+    status == 'finished'
+  end
+
+  def downloaded?
+    status == 'downloaded'
+  end
+
+  def failed?
+    status == 'failed'
+  end
+
+  def deleted?
+    status == 'deleted'
+  end
+
+  def email
+    if user.nil?
+      nil
+    else
+      user.email
+    end
+  end
+  
+  def filename
+    filename = "#{self.content}_#{formatted_create_at}.csv"
+  end
+  
+  def pending!
```

Comment:
how about only 1 method for status query and 1 for status setting ... 

```
def status?(x)
  x == :ready ? [:downloaded, :finished].include?(state) : state == x
end
```

also validate that only know states are being used